app/lib/frontend/backend.dart

```diff
@@ -159,12 +160,66 @@ class Backend {
     assert(repository.supportsDownloadUrl);
     return repository.downloadUrl(package, version);
   }
+
+  /// Stores a verification entry in the Datastore and returns its id that can
+  /// be used in client-communication, e.g. sending via e-mail.
+  ///
+  /// Duplicate detection relies on the serialized JSON form of [parameters],
+  /// and it is expected that callers will order the keys consistently.
+  /// When a duplicate is detected, the return value becomes `null`.
```

Comment:
What is the problem with duplicates?

If I try to add someone as uploader twice, I guess they get two emails and so what?

Reply:
I'm asking because I might be missing somewhere else where duplication is a problem :)

But isn't it better to ensure that the handlers that do something after verifying the secret is idempotent?

Reply:
The verification can be abused and send 100s of emails to the same address. I guess it can be abused in many other ways, e.g. sending separate emails to 100s of different addresses too, but at least the repeated sending to one email is filtered with this one easily.

E.g. I imagine when this is live, and someone adds a new uploader, and won't see it on the pub site right away, they will repeat the command many times.